app/blueprints/base/views.py

```diff
@@ -72,27 +73,54 @@ def authenticated_within(max_age):
 
 
 def force_authentication(path=None):
-    request_path = request.full_path
+    request_path = '/'
     if path is not None:
         request_path = path
     session["next_url"] = request_path
-    return redirect(oidc.login("dex", force_reauthentication=True))
+    return redirect(oidc.login(get_oidc_provider(),
+                               force_reauthentication=True))
 
 
+def get_oidc_provider():
+    oidc_provider = request.cookies.get('oidc_provider')
+    if oidc_provider is None:
+        oidc_provider = oidc.oidc_providers()[0]
+
+    return oidc_provider
+
+
+def set_oidc_provider_cookie(response, oidc_provider):
+    response.set_cookie('oidc_provider', oidc_provider)
+
+
+@base.route('/switch_oidc_provider/<path:caller>', methods=['GET', 'POST'])
```

Comment:
You're passing the view name in caller, not a path. Don't do this. It exposes implementation details in the URL unnecessarily and potentially opens up a URL injection attack. Instead, use `request.full_path` or `url_for('base.view')` - it doesn't expose any new information and you won't have to strip the `base` or the `.` later.